### PR TITLE
Add a `go mod tidy` check during CI

### LINKED ISSUE #276: Include a tidy check on the CI
Add some CI logic to check whether the `go.mod` and `go.sum` is consistent. The ideal would be something like the discussed here: https://github.com/golang/go/issues/27005

but it can also be achieved easily with some workaround for now, for example: https://github.com/grafana/grafana/pull/27204 and https://github.com/Kong/kubernetes-ingress-controller/pull/949/

### PR DESCRIPTION
**Pull request by jonathanbeber**
fixes #276

### CODE REVIEW

**Comment by jonathanbeber** [comment]
it works!
image

**Comment by jonathanbeber** [comment]
:+1:

**Comment by mikkeloscar** [comment]
:+1:

File: delivery.yaml [hunk 1]
```diff
@@ -13,6 +13,12 @@ pipeline:
   commands:
   - desc: test
     cmd: |
+      go mod tidy
+      STATUS=$( git status --porcelain go.mod go.sum )
+      if [ ! -z "$STATUS" ]; then
```
Comment (aermakov-zalando) [inline delivery.yaml:18]: `if ! git diff --stat --exit-code`?

Comment (jonathanbeber) [reply delivery.yaml:18]: great!

**Comment by jonathanbeber** [comment]
:+1:

**Comment by aermakov-zalando** [comment]
👍

**Pull request merged**